flag_int64_slice.go

```diff
@@ -135,7 +141,52 @@ func lookupInt64Slice(name string, set *flag.FlagSet) []int64 {
 		if err != nil {
 			return nil
 		}
+		// extract default value from the flag
+		var defaultVal []int64
+		for _, v := range strings.Split(f.DefValue, ",") {
+			iV, _ := strconv.ParseInt(v, 10, 64)
+			defaultVal = append(defaultVal, iV)
+		}
+		// if the current value is not equal to the default value
+		// remove the default values from the flag
+		if !isInt64SliceEqual(parsed, defaultVal) {
+			for _, v := range defaultVal {
+				parsed = removeFromInt64Slice(parsed, v)
+			}
+		}
 		return parsed
 	}
 	return nil
 }
+
+func removeFromInt64Slice(slice []int64, val int64) (newVal []int64) {
```

Comment:
Nit: We shouldn't be using named returns unless the function signature is ambiguous, or we absolutely have to

There are caveats with the scope of a named variable that aren't always intuitive, and naked returns pretty much always obfuscate what code should be doing. It also encourages us to mutate an existing variable rather than returning directly.

I'd probably write the function like this:
```go
func removeFromInt64Slice(slice []int64, val int64) ([]int64) {
	for i, v := range slice {
		if v == val {
			return slice[count+1:]
		}
	}
	return slice
}
```

That said, there's an issue here, which I'm not sure if is intentional or not. This function removes all items up to and including the first occurrence of `val`, but the name implies that we would just be removing `val`. I don't think it manifests because of how this function is called, but we should either fix the implementation or rename the function to avoid future misuse of this helper.

Reply:
I have updated the function to behave like it's named

```go
func removeFromInt64Slice(slice []int64, val int64) []int64 {
	for i, v := range slice {
		if v == val {
			return append(slice[:i], slice[i+1:]...)
		}
	}
	return slice
}
```

See it in action: https://play.golang.org/p/ZjJxp38tKNQ